File: Find Closest Number to Zero - Leetcode 2239/Find Closest Number to Zero - Leetcode 2239.cpp
#include <vector>
#include <cmath>
#include <algorithm>

class Solution {
public:
    int findClosestNumber(const std::vector<int>& nums) {
        int closest = nums[0];

        for (int x : nums) {
            if (std::abs(x) < std::abs(closest)) {
                closest = x;
            }
        }

        if (closest < 0 && std::find(nums.begin(), nums.end(), std::abs(closest)) != nums.end()) {
            return std::abs(closest);
        } else {
            return closest;
        }
    }
};

File: Find Closest Number to Zero - Leetcode 2239/Find Closest Number to Zero - Leetcode 2239.go
package solution

func abs(a int) int {
	if a < 0 {
		return a * -1
	}
	return a
}

func findClosestNumber(nums []int) int {
    closeNum := nums[0]
    for i:=1; i<len(nums); i++{        
        if abs(nums[i]) < abs(closeNum){
            closeNum = nums[i]
        }else if abs(nums[i]) == abs(closeNum) && nums[i] > closeNum{
            closeNum = nums[i]
        }
    }
    return closeNum
}
